packages/app/hooks/user/useUpdateUser.ts

```diff
@@ -1,18 +1,16 @@
-import { useMutation } from '@tanstack/react-query';
-import axios from 'axios';
-import { api } from 'app/constants/api';
 import { useUserSetter } from 'app/auth/hooks';
+import { queryTrpc } from 'app/trpc';
 
 export const useUpdateUser = () => {
-  const { mutateAsync: mutateUserAsync } = useMutation({
-    mutationFn: updateUserMutation,
-  });
+  const { mutateAsync: mutateUserAsync } = queryTrpc.editUser.useMutation();
   const setUser = useUserSetter();
 
   const updateUser = (user) => {
     (async () => {
       try {
-        const updatedUser = await mutateUserAsync(user);
+        const { userId, ...rest } = user;
+        const updatedUser = await mutateUserAsync({ id: userId, ...rest });
+        // @ts-ignore
```

Comment:
We should always avoid disabling linter in code.

I recommend that you fix the linting problem or add more comments on why you needed to disable lint here.

All it takes is for one developer to start disabling linting (perhaps for a good reason) for other developers to start doing the same thing everywhere, but for lazy reasons :) 

``suggestion
```